url5583f3dbb2_XOR19_XScript_tgz-pJ8-xscript_interactive_MainJ8/src/xscript/compiler/XTreeCompiler.java

```diff
@@ -0,0 +1,1298 @@
+						return;
+					} else {
+
+					}
+				} else {
+					if (hadSetup) {
+						code.addInstruction(operator, XOpcode.JUMP_IF_ZERO, target);
+					}
+					hadSetup = true;
+					part.setupAndGet(this, code);
+				}
+				part = visit(operator.exprs.get(i));
+			}
+			if (part.isConstValue()) {
+				if (part.isFalse()) {
+					// TODO
+					treePart = new XCompiledPartConst(operator, false);
+					return;
+				}
+			} else {
+				if (hadSetup) {
+					code.addInstruction(operator, XOpcode.JUMP_IF_ZERO, target);
+				}
+				part.setupAndGet(this, code);
+			}
+			XJumpTarget t = new XJumpTarget();
+			code.addInstruction(operator, XOpcode.JUMP, t);
+			code.addInstruction(operator, target);
+			code.addInstruction(operator, XOpcode.LOAD_FALSE);
+			code.addInstruction(operator, t);
+		} else {
+			XCodeGen code = getCodeExpr(operator);
+			XCompiledPart part = visit(operator.exprs.get(0));
+			int i = 1;
+			Object _const = null;
+			boolean isConst = part.isConstValue();
+			if (isConst) {
+				_const = part.getConst();
+			} else {
+				part.setupAndGet(this, code);
+			}
+			for (XOperator oo : operator.operators) {
+				part = visit(operator.exprs.get(i++));
+				if (isConst && part.isConstValue()) {
+					try {
+						_const = oo.calc(_const, part.getConst());
+					} catch (UnsupportedOperationException e) {
+						addDiagnostic(operator, "op.not.supported", oo, _const == null ? null : _const.getClass(), part.getConst() == null ? null : part.getConst().getClass());
+						code.addInstruction(operator, XOpcode.LOADN);
+						isConst = false;
+					}
+				} else {
+					if (isConst) {
+						code.addInstruction(new XInstConst(operator.position.position.line, _const));
+						isConst = false;
+					}
+					part.setupAndGet(this, code);
+					code.addInstruction(operator, oo.opcode);
+				}
+			}
+			if (isConst) {
+				treePart = new XCompiledPartConst(operator, _const);
+			}
+		}
+	}
+
+	@Override
+	public void visitOperatorIf(XTreeOperatorIf operatorIf) {
+		XCodeGen code = getCodeExpr(operatorIf);
+		XCompiledPart part = visit(operatorIf.condition);
+		if (part.isConstValue()) {
+			if (part.isFalse()) {
+				addDiagnostic(operatorIf.onTrue, "dead.code");
+				visit(operatorIf.onFalse).setupAndGet(this, code);
+			} else {
+				visit(operatorIf.onTrue).setupAndGet(this, code);
+				addDiagnostic(operatorIf.onFalse, "dead.code");
+			}
+		} else {
+			part.setupAndGet(this, code);
+			XJumpTarget onFalse = new XJumpTarget();
+			code.addInstruction(operatorIf, XOpcode.JUMP_IF_ZERO, onFalse);
+			visit(operatorIf.onTrue).setupAndGet(this, code);
+			XJumpTarget end = new XJumpTarget();
+			code.addInstruction(operatorIf, XOpcode.JUMP, end);
+			code.addInstruction(operatorIf, onFalse);
+			visit(operatorIf.onFalse).setupAndGet(this, code);
+			code.addInstruction(operatorIf, end);
+		}
+	}
+
+	@Override
+	public void visitOperatorPrefixSuffix(XTreeOperatorPrefixSuffix operatorPrefixSuffix) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 24. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 941-967 into a separate method.